Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp

```diff
@@ -221,14 +221,11 @@ String CryptoDigest::toHexString()
 {
     auto hash = computeHash();
 
-    char* start = 0;
+    std::span<char> start;
     unsigned hashLength = hash.size();
     CString result = CString::newUninitialized(hashLength * 2, start);
-    char* buffer = start;
-    for (size_t i = 0; i < hashLength; ++i) {
-        snprintf(buffer, 3, "%02X", hash.at(i));
-        buffer += 2;
-    }
+    for (size_t i = 0; i < hashLength; ++i)
+        snprintf(start.subspan(i * 2).data(), 3, "%02X", hash.at(i));
```

Comment:
This copied and pasted code should probably share a function with the one from `SHA1.cpp`. The same mistakes are here. But also here seems unnecessary to use a `CString` and turn it into a `String`. More efficient to make a Latin-1 `String` directly. So maybe need a separate `String` and `CString` version?